src/services/stats-wales-api.ts

```diff
@@ -114,70 +114,74 @@ export class StatsWalesApi {
+        logger.debug(`Confirming file import: ${factTableId}`);
 
         return this.fetch({
-            url: `dataset/${datasetId}/revision/by-id/${revisionId}/import/by-id/${importId}/confirm`,
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/fact-table/by-id/${factTableId}/confirm`,
             method: HttpMethod.Patch
-        }).then((response) => response.json() as unknown as FileImportDTO);
+        }).then((response) => response.json() as unknown as FactTableDto);
     }
 
     public async getSourcesForFileImport(
         datasetId: string,
         revisionId: string,
-        importId: string
-    ): Promise<FileImportDTO> {
-        logger.debug(`Fetching sources for file import: ${importId}`);
+        factTableId: string
+    ): Promise<FactTableDto> {
+        logger.debug(`Fetching sources for file import: ${factTableId}`);
 
         return this.fetch({
-            url: `dataset/${datasetId}/revision/by-id/${revisionId}/import/by-id/${importId}`
-        }).then((response) => response.json() as unknown as FileImportDTO);
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/fact-table/by-id/${factTableId}`
+        }).then((response) => response.json() as unknown as FactTableDto);
     }
 
-    public async removeFileImport(datasetId: string, revisionId: string, importId: string): Promise<DatasetDTO> {
-        logger.debug(`Removing file import: ${importId}`);
+    public async removeFileImport(datasetId: string, revisionId: string, factTableId: string): Promise<DatasetDTO> {
+        logger.debug(`Removing file import: ${factTableId}`);
 
         return this.fetch({
-            url: `dataset/${datasetId}/revision/by-id/${revisionId}/import/by-id/${importId}`,
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/fact-table/by-id/${factTableId}`,
             method: HttpMethod.Delete
         }).then((response) => response.json() as unknown as DatasetDTO);
     }
 
     public async getImportPreview(
         datasetId: string,
         revisionId: string,
-        importId: string,
+        factTableId: string,
         pageNumber: number,
         pageSize: number
     ): Promise<ViewDTO> {
         logger.debug(
-            `Fetching preview for dataset: ${datasetId}, revision: ${revisionId}, import: ${importId}, page: ${pageNumber}, pageSize: ${pageSize}`
+            `Fetching preview for dataset: ${datasetId}, revision: ${revisionId}, import: ${factTableId}, page: ${pageNumber}, pageSize: ${pageSize}`
         );
 
         return this.fetch({
-            url: `dataset/${datasetId}/revision/by-id/${revisionId}/import/by-id/${importId}/preview?page_number=${pageNumber}&page_size=${pageSize}`
+            url: `dataset/${datasetId}/revision/by-id/${revisionId}/fact-table/by-id/${factTableId}/preview?page_number=${pageNumber}&page_size=${pageSize}`
```

Comment:
Do we think this `by-id` stuff is still valuable since the schema change? If so, fine, otherwise let's strip it out in a future refactor.

Reply:
I started sketching out the API on the my miro board the other day... and came up with this.  For revisions you'll want by-index/version and ID.  Dataset you'll probably want to get by URLized title or code, we've not talked about code recently...  but its the codes like `tran0303`... That'll play into "Don't break the internet"... I believe its probably still a thing, I suspect the sticking point will be how those codes are generated...  While I'd like a simpler route at least `by-id` explains what you're doing.

If you want to suggest an alternative route layout I'm all for it.  I have other ideas but until MVP lands I think its something we should stick in the tech-debt pile to talk about and agree later...  I'd even put a date on it of the 1st of February to talk about it and agree which way we want to go.